src/OSPSuite.Core/Domain/Services/ModelConstructor.cs

```diff
@@ -112,6 +113,20 @@ private void finalizeModel(IModel model)
 
          //now we should be able to resolve all references
          _referencesResolver.ResolveReferencesIn(model);
+
+
+         //This should be done after reference were resolved to ensure that we do not remove formula parameter that could not be evaluated
+         removeUndefinedInstanceParametersIn(model);
+      }
+
+      private void removeUndefinedInstanceParametersIn(IModel model)
+      {
+         var allNaNParametersFromMolecules = model.Root.GetAllChildren<IContainer>()
+            .Where(c => c.ContainerType == ContainerType.Molecule)
```

Comment:
@Yuri05  Right now only for Molecule. But..should we do this for all parameters in a simulation

@PavelBal Can you think of a use case where it makes sense to have parameter in a simulation with a NaN value?

Reply:
@msevestre No, actually not. Get rid of them :)

Reply:
>But..should we do this for all parameters in a simulation?

No. Even though I do not have any use case in mind where we would need NaN-Parameter: they do not harm. So please keep them.